src/utils/workspaceUtils.ts

```diff
@@ -47,49 +46,39 @@ export async function selectWorkspaceItem(context: IActionContext, placeHolder:
 }
 
-export async function getWorkspaceFolder(context: IActionContext): Promise<WorkspaceFolder> {
-    let folder: WorkspaceFolder | undefined;
+export async function tryGetWorkspaceFolder(context: IActionContext): Promise<WorkspaceFolder | undefined> {
     context.telemetry.properties.noWorkspaceResult = 'canceled';
-
     if (!workspace.workspaceFolders || workspace.workspaceFolders.length === 0) {
-        const noWorkspaceWarning: string = 'noWorkspaceWarning';
-        const message: string = localize(noWorkspaceWarning, 'You must have a git project open to create a Static Web App.');
-        const cloneProject: MessageItem = { title: localize('cloneProject', 'Clone project from GitHub') };
-        const openExistingProject: string = 'openExistingProject';
-        const openExistingProjectMi: MessageItem = { title: localize(openExistingProject, 'Open existing project') };
-        const result: MessageItem = await context.ui.showWarningMessage(message, { modal: true, stepName: noWorkspaceWarning }, openExistingProjectMi, cloneProject);
-
-        if (result === cloneProject) {
-            await cloneRepo(context, '');
-            context.telemetry.properties.noWorkspaceResult = 'cloneProject';
-        } else {
-            const uri: Uri[] = await context.ui.showOpenDialog({
-                canSelectFiles: false,
-                canSelectFolders: true,
-                canSelectMany: false,
-                openLabel: localize('open', 'Open'),
-                stepName: openExistingProject
-            });
-            // don't wait
-            void commands.executeCommand('vscode.openFolder', uri[0]);
-            context.telemetry.properties.noWorkspaceResult = openExistingProject;
-        }
-
-        context.errorHandling.suppressDisplay = true;
-        throw new NoWorkspaceError();
+        return;
     } else if (workspace.workspaceFolders.length === 1) {
-        folder = workspace.workspaceFolders[0];
         context.telemetry.properties.noWorkspaceResult = 'singleRootProject';
+        return workspace.workspaceFolders[0];
     } else {
-        const selectProjectFolder: string = 'selectProjectFolder';
-        const placeHolder: string = localize(selectProjectFolder, 'Select the folder containing your SWA project');
-        folder = await window.showWorkspaceFolderPick({ placeHolder });
+        const selectAppFolder: string = 'selectAppFolder';
+        const placeHolder: string = localize(selectAppFolder, 'Select folder with your app');
+        const folder = await window.showWorkspaceFolderPick({ placeHolder });
         if (!folder) {
-            throw new UserCancelledError(selectProjectFolder);
+            throw new UserCancelledError(selectAppFolder);
         }
-
         context.telemetry.properties.noWorkspaceResult = 'multiRootProject';
+        return folder;
     }
+}
+
+export async function showNoWorkspacePrompt(context: IActionContext): Promise<MessageItem> {
```

Comment:
I think we should handle setting the contexts here for whatever the user chooses.  This command is specific to `createStaticApp` so we shouldn't need to use the generic IActionContext for it (which is why I assumed that you didn't set the contexts here)

Reply:
Are you referring to the telemetry contexts? If you are, then I can still use IActionContext.

When I add the "Create from template" option, I will change it from the generic so that I can set the context inside this function for if the user picks "Create from template".

Reply:
Oh no,  I was referring to these lines:
https://github.com/microsoft/vscode-azurestaticwebapps/pull/466/files#diff-71e8df3edc0b95484e8ea0cfb8fb79ffc32b5160dd9e62fb6717d0e7ee711b07R36-R45

It's kind of weird to me that this isn't handled in the same function as the prompt.

Reply:
Ok, makes sense. I went ahead and moved those into the prompt, including `throw new NoWorkspaceError();`. When I implement the create from template option, I'll have to make sure to not throw that error when they select create from template.